include/seqan3/core/detail/debug_stream_byte.hpp

```diff
@@ -0,0 +1,43 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Enrico Seiler <enrico.seiler AT fu-berlin.de>
+ * \brief Provides seqan3::debug_stream and related types.
+ */
+
+#pragma once
+
+#include <cstddef>
+
+#include <seqan3/core/detail/debug_stream_type.hpp>
+#include <seqan3/std/concepts>
+
+namespace seqan3
+{
+/*!\name Formatted output overloads
+ * \{
+ */
+/*!\brief A std::byte can be printed by converting it to an integer.
+ * \tparam    byte_type     The type of the input; must be equal to `std::byte`.
+ * \param[in] s             The seqan3::debug_stream.
+ * \param[in] arg           The std::byte.
+ * \relates seqan3::debug_stream_type
+ */
+template <typename byte_type, typename char_t>
+//!\cond
+    requires std::same_as<remove_cvref_t<byte_type>, std::byte>
+//!\endcond
+inline debug_stream_type<char_t> & operator<<(debug_stream_type<char_t> & s, byte_type && arg)
+{
+    s << static_cast<uint8_t>(arg);
```

Comment:
```suggestion
    s << std::to_integer<size_t>(arg);
```
?
That would guarantee that we see actually the number, right? Otherwise it could happen, that the byte is interpreted as char on some platforms that would print nothing for low values.